### ISSUE TITLE
Channel search returns undefined name 

### ISSUE THREAD

**Issue by KennethWussmann** [d-fischer/twitch#249]
## Bug Report
Hey,
I'm using the below code to search for channels on Twitch and I'm interested in the login username (not the display name) of the returned `HelixUser[]`.
Thanks for your help!

### Code

```ts
const channelsResponse = await this.apiClient.helix.search.searchChannels("ludwig", {
  limit: '10',
  after: undefined
});
const firstChannel: HelixUser = channelsResponse.data[0];
console.log("Name:", firstChannel.name);
console.log("Display name:", firstChannel.displayName);
// prints:
// Name: undefined
// Display name: ludwig
```

### Expected behavior

[`HelixUser.name`](https://d-fischer.github.io/twitch/reference/classes/HelixUser.html#name) returns `ludwig`.

Because according to the docs and TypeScript definition, [`HelixUser.name`](https://d-fischer.github.io/twitch/reference/classes/HelixUser.html#name) is of non-nullable type `string`.

### Actual Behavior

[`HelixUser.name`](https://d-fischer.github.io/twitch/reference/classes/HelixUser.html#name) returns `undefined`.

It's also not just this search query, it's a general issue.

I'm not sure how close you implemented the fields according to the Twitch API responses, but the property `name` does not exist in the Twitch API response: https://dev.twitch.tv/docs/api/reference#search-channels

I think the desired property would be `broadcaster_login`. That's also already filled in the protected `_data: HelixUserData` of `HelixUser` with my desired value, but it's not exposed to the public API: 

![Bildschirmfoto 2021-03-29 um 23 07 57](https://user-images.githubusercontent.com/11491506/112900310-9c593500-90e3-11eb-8a12-033fa0832a6b.png)

### Environment

- Version: `4.5.2`
- Node version: `15.2.0`
- Operating system: `macOS Big Sur 11.2.2 (Intel)`

**Comment by d-fischer** [comment]
Good catch! I actually wrote the class for channel search results, but for some reason I ended up using `HelixUser` anyway.

After this is fixed, you should be able to keep using the `.name` property and get the correct result.

On a slightly related note, you can just use `.helix.users.getUserByName('ludwig')` and get more reliable results if you're looking for an exact name. (The exact match is not always the "best" match for a search according to Twitch for some reason)

**Comment by d-fischer** [comment]
Fixed by d2311e5.
